### PR TITLE
Add Validate() function for audit options

### PR DESCRIPTION
**Pull request by CaoShuFeng**
**Release note**:

```
NONE
```
Fixes: #47114

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot ok to test

File: staging/src/k8s.io/apiserver/pkg/server/options/audit.go [hunk 1]
```diff
@@ -147,14 +176,20 @@ func (o *AuditLogOptions) applyTo(c *server.Config) error {
 			MaxSize:    o.MaxSize,
 		}
 	}
-	c.LegacyAuditWriter = w
-
-	if advancedAuditingEnabled() {
+	return w
+}
+func (o *AuditLogOptions) applyToAdvanced(c *server.Config) error {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit.go]: empty line above

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit.go]: matter of taste: advancedApplyTo and legacyApply To

File: staging/src/k8s.io/apiserver/pkg/server/options/audit.go [hunk 2]
```diff
@@ -91,13 +118,18 @@ func (o *AuditOptions) AddFlags(fs *pflag.FlagSet) {
 }
 
 func (o *AuditOptions) ApplyTo(c *server.Config) error {
+	// Legacy audit
+	if !advancedAuditingEnabled() {
+		return o.LogOptions.applyToLegacy(c)
+	}
+
 	// Apply generic options.
 	if err := o.applyTo(c); err != nil {
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit.go]: can we inline applyTo's contents here? I think this is confusing otherwise.

**Comment by sttts** [comment]
Some nits.

File: staging/src/k8s.io/apiserver/pkg/server/options/audit.go (hunk 2 above)
Comment (CaoShuFeng) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit.go]: Done.

File: staging/src/k8s.io/apiserver/pkg/server/options/audit.go (hunk 1 above)
Comment (CaoShuFeng) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit.go]: Done.

**Comment by CaoShuFeng** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by CaoShuFeng** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**